### PR TITLE
Add account creation

### PR DESCRIPTION
**Pull request by wmontwe**
This collected the account setup steps and put them in one place to safely create a new account.

Depends on: #7036

### CODE REVIEW

File: app/k9mail/src/main/java/com/fsck/k9/account/AccountOwnerNameProvider.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.fsck.k9.account
+
+import app.k9mail.feature.account.setup.domain.ExternalContract
+import com.fsck.k9.Preferences
+
+class AccountOwnerNameProvider(
+    private val preferences: Preferences,
+) : ExternalContract.AccountOwnerNameProvider {
+    override fun getOwnerName(): String? {
+        return preferences.defaultAccount?.senderName
```
Comment (cketti) [inline app/k9mail/src/main/java/com/fsck/k9/account/AccountOwnerNameProvider.kt:10]: This call has the potential to perform disk I/O. So we should only call it from a background thread.

File: app/k9mail/src/main/java/com/fsck/k9/account/AccountSetupFinishedLauncher.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import app.k9mail.feature.account.setup.domain.ExternalContract
+import com.fsck.k9.Preferences
+import com.fsck.k9.activity.MessageList
+
+class AccountSetupFinishedLauncher(
+    private val context: Context,
+    private val preferences: Preferences,
+) : ExternalContract.AccountSetupFinishedLauncher {
+    override fun launch(accountUuid: String) {
+        val account = preferences.getAccount(accountUuid)
```
Comment (cketti) [inline app/k9mail/src/main/java/com/fsck/k9/account/AccountSetupFinishedLauncher.kt:13]: This call has the potential to perform disk I/O. So we should only call it from a background thread.

File: app/k9mail/src/main/java/com/fsck/k9/account/AccountCreator.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            newAccount.displayCount = account.options.messageDisplayCount
+
+            newAccount.folderPushMode = FolderMode.ALL // TODO is this always ALL?
+            newAccount.deletePolicy =
+                accountCreatorHelper.getDefaultDeletePolicy(newAccount.incomingServerSettings.type)
+            newAccount.chipColor = accountCreatorHelper.pickColor()
+
+            // TODO check this:
+            // DI.get(LocalKeyStoreManager.class)
+            //      .deleteCertificate(mAccount, newHost, newPort, MailServerDirection.OUTGOING)
```
Comment (cketti) [inline app/k9mail/src/main/java/com/fsck/k9/account/AccountCreator.kt:47]: Deleting old certificates is only necessary when updating server settings of an existing account. Not when creating a new account.

File: feature/account/setup/src/test/kotlin/app/k9mail/feature/account/setup/ui/options/AccountOptionsStateMapperKtTest.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+class AccountOptionsStateMapperKtTest {
+
+    @Test
+    fun `should map state to account options`() {
+        val state = AccountOptionsContract.State(
+            accountName = StringInputField("accountName"),
+            displayName = StringInputField("displayName"),
+            emailSignature = StringInputField("emailSignature"),
+            checkFrequency = EmailCheckFrequency.Companion.DEFAULT,
+            messageDisplayCount = EmailDisplayCount.DEFAULT,
```
Comment (cketti) [inline feature/account/setup/src/test/kotlin/app/k9mail/feature/account/setup/ui/options/AccountOptionsStateMapperKtTest.kt:20]: Using the default values here implicitly checks the implementation of `EmailCheckFrequency` and `EmailDisplayCount`. But you shouldn't have to touch this test when changing the defaults in either of those classes.

**Review by cketti — APPROVED** [review]

File: feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/ui/AccountSetupViewModel.kt [hunk 5]
```diff
@@ -56,11 +73,29 @@ class AccountSetupViewModel(
         }
     }
 
-    private fun navigateNext() {
-        // TODO: validate account
+    private fun onFinish() {
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/ui/AccountSetupViewModel.kt:76]: So far most method calls triggered from an event have been named onEventName.

**Pull request merged**